src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp

```diff
@@ -0,0 +1,323 @@
+
+    error = CHIP_NO_ERROR;
+exit:
+    if (asymm_ctx.session != NULL)
+    {
+        sss_asymmetric_context_free(&asymm_ctx);
+    }
+
+    return error;
+}
+
+CHIP_ERROR P256KeypairHSM::Serialize(P256SerializedKeypair & output)
+{
+    CHIP_ERROR error = CHIP_ERROR_INTERNAL;
+    size_t len       = output.Length() == 0 ? output.Capacity() : output.Length();
+    Encoding::BufferWriter bbuf(output, len);
+    uint8_t privkey[kP256_PrivateKey_Length];
+
+    {
+        /* Set the public key */
+        P256PublicKey & public_key = const_cast<P256PublicKey &>(Pubkey());
+        bbuf.Put(Uint8::to_uchar(public_key), public_key.Length());
+    }
+
+    VerifyOrExit(bbuf.Available() == sizeof(privkey), error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(sizeof(privkey) >= 4, error = CHIP_ERROR_INTERNAL);
+
+    {
+        /* When HSM is used for ECC key generation, store key info in private key buffer */
+        Encoding::LittleEndian::BufferWriter privkey_bbuf(privkey, sizeof(privkey));
+        privkey_bbuf.Put32(keyid);
+    }
+
+    bbuf.Put(privkey, sizeof(privkey));
+    VerifyOrExit(bbuf.Fit(), error = CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    output.SetLength(bbuf.Needed());
+
+    error = CHIP_NO_ERROR;
+exit:
+    return error;
+}
+
+CHIP_ERROR P256KeypairHSM::Deserialize(P256SerializedKeypair & input)
+{
+    CHIP_ERROR error = CHIP_ERROR_INTERNAL;
+
+    /* Set the public key */
+    const P256PublicKey & public_key = Pubkey();
+    Encoding::BufferWriter bbuf((uint8_t *) Uint8::to_const_uchar(public_key), public_key.Length());
+
+    VerifyOrExit(input.Length() == public_key.Length() + kP256_PrivateKey_Length, error = CHIP_ERROR_INVALID_ARGUMENT);
+    bbuf.Put(static_cast<uint8_t *>(input), public_key.Length());
+
+    /* Set private key info */
+    VerifyOrExit(bbuf.Fit(), error = CHIP_ERROR_NO_MEMORY);
+    {
+        /* When HSM is used for ECC key generation, key info in stored in private key buffer */
+        const uint8_t * privkey = Uint8::to_const_uchar(input) + public_key.Length();
+        keyid                   = Encoding::LittleEndian::Get32(privkey);
+    }
+
+    error = CHIP_NO_ERROR;
+exit:
+    return error;
+}
+
+CHIP_ERROR P256KeypairHSM::ECDH_derive_secret(const P256PublicKey & remote_public_key, P256ECDHDerivedSecret & out_secret) const
+{
+    CHIP_ERROR error           = CHIP_ERROR_INTERNAL;
+    const uint8_t * rem_pubKey = NULL;
+    size_t rem_pubKeyLen       = 0;
+    size_t secret_length       = (out_secret.Length() == 0) ? out_secret.Capacity() : out_secret.Length();
+    smStatus_t smstatus        = SM_NOT_OK;
+
+    VerifyOrExit(keyid != 0, error = CHIP_ERROR_HSM);
+
+    ChipLogDetail(Crypto, "ECDH_derive_secret: Using SE05X for ECDH !");
+
+    se05x_sessionOpen();
+
+    rem_pubKey    = Uint8::to_const_uchar(remote_public_key);
+    rem_pubKeyLen = remote_public_key.Length();
+
+    VerifyOrExit(rem_pubKey != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
both inputs are passed as reference so these checks are not required

Reply:
removed the checks